pkg/drivers/nats/new.go

```diff
@@ -93,6 +94,16 @@ func newBackend(ctx context.Context, wg *sync.WaitGroup, connection string, tlsI
 
 		// Start the server.
 		go ns.Start()
+
+		// Shutdown on interrupt.
+		sigch := make(chan os.Signal, 1)
+		signal.Notify(sigch, os.Interrupt)
+		go func() {
+			<-sigch
+			ns.Shutdown()
+			logrus.Infof("embedded NATS server shutdown")
+		}()
```

Comment:
No, don't do your own signal handling. This is handled by the CLI or calling application:
* https://github.com/k3s-io/kine/blob/1d571b01fd1e5e08b0276f7d06c3244972acb773/pkg/app/app.go#L225
* https://github.com/k3s-io/kine/blob/1d571b01fd1e5e08b0276f7d06c3244972acb773/pkg/signals/signal.go#L34-L37

You should be using the context and waitGroup that you are provided - call wg.Add(1) when starting, do what you're going to do until the context is done, and then call wg.Done() when you've shut down.